src/transformers/modeling_rope_utils.py

```diff
@@ -189,13 +189,31 @@ def _compute_yarn_parameters(
     partial_rotary_factor = config.partial_rotary_factor if hasattr(config, "partial_rotary_factor") else 1.0
     head_dim = getattr(config, "head_dim", config.hidden_size // config.num_attention_heads)
     dim = int(head_dim * partial_rotary_factor)
```

Comment:
A problem is that `dim` is wrong for the DeepSeek model, it has to be `config.qk_rope_head_dim`. I'd suggest this (sorry, cannot comment on parts of the code not changed):

```
def _compute_yarn_parameters(
    config: PretrainedConfig,
    device: "torch.device",
    seq_len: Optional[int] = None,
    dim: Optional[int] = None,
    **rope_kwargs
) -> Tuple["torch.Tensor", float]:
```

Then, replace line 191 with:
```
if dim is None:
    dim = int(head_dim * partial_rotary_factor)
```

Now, we can call it with the correct `dim`.

Reply:
I take care of this [here](https://github.com/huggingface/transformers/pull/35926/files#diff-fe48f18f563aebc9846599f55407d41815a60e52fb85b036edc2003e229e7863R205)!

in configuration file,
`self.head_dim = qk_rope_head_dim`

Reply:
I would not recommend that. `self.head_dim` is used in other places, it should remain independent of RoPE. In fact, the correct `head_dim` for DeepSeek models would be `qk_nope_head_dim + qk_rope_head_dim`.